microservice/subscription/src/test/java/com/cumulocity/microservice/subscription/repository/impl/LegacyMicroserviceRepositoryTest.java

```diff
@@ -320,6 +322,102 @@ public void shouldGetSubscriptions(){
                 .isSameAs(user);
     }
 
+    @Test
+    public void shouldCreateAppWithExtensionsIfMetadataContains() {
+        repository.register(
+                getMetadataWithExtensions());
+
+        assertThat(platform.take(byMethod(POST)))
+                .hasSize(1)
+                .extracting("body")
+                .have(new Condition<Object>() {
+                    @Override
+                    public boolean matches(Object value) {
+                        if (value instanceof ApplicationRepresentation) {
+                            assertThat(((ApplicationRepresentation) value).get(MicroserviceMetadataRepresentation.EXTENSIONS_FIELD_NAME)).isNotNull();
+                            return true;
+                        }
+                        return false;
+                    }
+                });
+        assertThat(platform.take(byMethod(PUT))).isEmpty();
+    }
+
+    @Test
+    public void shouldUpdateAppWithExtensionsIfMetadataContains() {
+        ApplicationRepresentation existing = applicationRepresentation()
+                .id("3")
+                .type(MICROSERVICE)
+                .name(CURRENT_APPLICATION_NAME)
+                .build();
+        givenApplications(existing);
+
+        repository.register(getMetadataWithExtensions());
+
+        assertThat(platform.take(byMethod(PUT)))
+                .hasSize(1)
+                .extracting("body")
+                .have(new Condition<Object>() {
```

Comment:
this is exactly the same condition as above, another reason to extract it to a method